test/controllers/classrooms_controller_test.rb

```diff
@@ -1,66 +1,73 @@
 require "test_helper"
 
 class ClassroomsControllerTest < ActionDispatch::IntegrationTest
-  setup do
-    @classroom = classrooms(:one)
-    @user = users(:one) # Assuming you have a user fixture named :one
-    sign_in @user # S
-  end
```

Comment:
In my opinion, tests should be explicit. Having setup code like this makes it harder to understand. DRY should not apply in tests.